image-registry/automate-pruning.md

```diff
@@ -0,0 +1,267 @@
+---
+title: Automate-Pruning
+authors:
+  - "@dmage"
+  - "@adambkaplan"
+reviewers:
+  - "@bparees"
+  - "@coreydaley"
+  - "@ricardomarascini"
+  - "@gabemontero"
+approvers:
+  - "@bparees"
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2019-09-26
+last-updated: 2019-10-08
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Automate Image Pruning
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This enhancement will install a `CronJob` to automate the image pruning task
+for the internal registry. An API will be exposed to allow cluster admins to
+configure the pruning job to suit their needs, or disable pruning if that is
+what is desired.
+
+## Motivation
+
+Regular image pruning is a vital maintenance task to ensure an OpenShift
+cluster remains healthy. In certain use cases (such as CI/build farms), pruning
+must be run frequently to prevent etcd from exhausting storage. Today, pruning
+is configured through a manual, multi-step process
+
+### Goals
+
+- Provide a one-step process for cluster admins to regularly schedule image
+pruning.
+- Default pruning to be enabled on an OpenShift cluster.
+- Provide alerting if a cluster is at risk due to pruning failing/too many
+images.
+
+### Non-Goals
+
+- Fix fundamental issues with the current pruning process (ex: pruning images
+that may be referenced in a CRD).
+
+## Proposal
+
+### User Stories
+
+#### Prune images in etcd
+
+As an OpenShift cluster admin
+I want the cluster to automatically prune ImageStreams stored in etcd
+So that etcd storage is not overwhelmed by unused imagestreams
+And the cluster remains healthy
+
+#### Prune images on the registry
+
+As an OpenShift cluster admin
+I want the cluster to automatically prune unused images stored in the registry
+So that registry storage is managed
+And I don’t pay for storage consumed by images my organization doesn’t use
+
+#### Orphaned blobs
+
+As an OpenShift cluster admin
+I want the cluster to automatically remove orphaned blobs stored in the registry
+So that registry storage is managed
+And I don’t pay for storage consumed by images my organization doesn’t use
+
+### Implementation Details/Notes/Constraints
+
+The implementation relies the current image pruning process invoked by the
```

Comment:
`relies the current image pruning process` -> `relies on the current image pruning process`

i.e. add the `on1